### PR TITLE
Add clippy to workflows + Solve remaining clippy warnings

### PR DESCRIPTION
**Pull request by fmoletta**
Solves #26 & #5 
Depends on #35 and #32

### CODE REVIEW

**Comment by Oppen** [comment]
Why does a workflow PR depend on functionality ones?

**Comment by fmoletta** [comment]
> Why does a workflow PR depend on functionality ones?

It depends on #32 as it solves problems between cairo-rs and cairo-rs-py, allowing the main branch of cairo-rs to be used when building cairo-rs-py.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      uses: actions-rs/toolchain@v1
+      with:
+          toolchain: 1.61.0
+          override: true
+          components: rustfmt, clippy
+    - uses: actions/checkout@v3
+    - name: Python3 Build
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10.8'
```
Comment (Oppen) [inline .github/workflows/rust.yml:26]: I thought we were using 3.8.x as reference? @Juan-M-V I seem to recall you mentioned this not too long ago.

**Review by Oppen — APPROVED** [review]
Changes relevant to the workflow itself LGTM. I didn't re-review the ones from the other PR.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (juanbono) [reply .github/workflows/rust.yml:26]: Last monday I tried to setup the environment with 3.8.x and it fails with an error (print is not defined). But it works with Python 3.10.8.

Comment (Oppen) [reply .github/workflows/rust.yml]: I think @entropidelic found a fix for that.

Comment (fmoletta) [reply .github/workflows/rust.yml]: Yes! with the fix merged python version was changed to 3.9 (matching cairo-rs workflow)

**Review by Juan-M-V — APPROVED** [review]

**Pull request merged**